SYCL/ESIMD/api/functional/functions/functions_atomic_update_inc_dec.cpp

```diff
@@ -0,0 +1,230 @@
+                                (i - NumberIteractions) * num_changed_elems;
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "atomic_update return value");
+        }
+      }
+    }
+
+    std::vector<size_t> selected_indexes;
+    // Collect the indexess that has been selected.
+    for (size_t i = 0; i < NumElemsToChange; ++i) {
+      if (filter(i) == 1) {
+        selected_indexes.push_back(indexes_to_change[i]);
+      }
+    }
+
+    // Push the largest value to avoid the following error: can't dereference
+    // out of range vector iterator.
+    selected_indexes.push_back(std::numeric_limits<size_t>::max());
+    auto next_selected_index = selected_indexes.begin();
+
+    const DataT &expected = base_value;
+    DataT expected_after_change = functions::get_expected_value<ChosenOperator>(
+        base_value, NumberIteractions);
+    // Verify that values, that do not was changed has initial values.
+    for (size_t i = 0; i < NumElems; ++i) {
+      // If current index is less than selected index verify that this element
+      // hasn't been changed.
+      const DataT &retrieved = shared_init_values[i];
+      if (i < *next_selected_index) {
+        if (expected != retrieved) {
+          passed = fail_test(i, expected, retrieved, data_type,
+                             "value that should be updated");
+        }
+      } else {
+        if (expected_after_change != retrieved) {
+          passed = fail_test(i, expected_after_change, retrieved, data_type,
+                             "value that should be updated");
+        }
+        next_selected_index++;
+      }
+    }
+
+    return passed;
+  }
+
+private:
+  bool fail_test(size_t i, DataT expected, DataT retrieved,
+                 const std::string &data_type,
+                 const std::string &verification_type) {
+    log::fail(TestDescriptionT(data_type), "Unexpected value at index ", i,
+              ", retrieved: ", retrieved, ", expected: ", expected,
+              ", for verification type: ", verification_type,
+              ", with number elements to change: ", NumElemsToChange);
+
+    return false;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue queue(esimd_test::ESIMDSelector{},
+                    esimd_test::createExceptionHandler());
+
+  bool passed = true;
+
+  const auto uint_types = named_type_pack<uint32_t>::generate("uint32_t");
+  const auto single_size = get_sizes<16>();
+  const auto nd_range_dims = integer_pack<1, 2, 3>::generate_unnamed();
+  const auto num_elems_to_change = integer_pack<1, 8>::generate_unnamed();
+
+  const auto filter_types =
+      unnamed_type_pack<functions::filters::ChangeByStep,
+                        functions::filters::ChangeAll,
+                        functions::filters::ChangeNothing>::generate();
+  const auto atomic_op_types =
+      value_pack<atomic_op, atomic_op::inc, atomic_op::dec>::generate_unnamed();
```

Comment:
Feel free to provide `StringMaker` specialization for `esimd::atomic_op` and use something like:
```suggestion
  const auto atomic_op_types =
      value_pack<atomic_op, atomic_op::inc, atomic_op::dec>::generate_named();
```
or
```suggestion
  const auto atomic_op_types =
      atomic_op_pack<atomic_op::inc, atomic_op::dec>::generate_named();
```

You would be able to clarify in logs which operation failed by that.

Reply:
I've applied this comment in this commit: https://github.com/intel/llvm-test-suite/pull/939/commits/ab63a6d73d8414bb25b22058560e708893f02f35